include/seastar/json/json_elements.hh

```diff
@@ -186,19 +189,27 @@ public:
      * iteration and that it's elements can be assigned to the list elements
      */
     template<class C>
-    json_list& operator=(const C& list) {
+    json_list_template& operator=(const C& list) {
         _elements.clear();
         for  (auto i : list) {
             push(i);
         }
         return *this;
     }
     virtual future<> write(output_stream<char>& s) const override {
-        return formatter::write(s, _elements);
+        auto t = const_cast<json_list_template<T, Container> *>(this);
```

Comment:
I'm puzzled by this cast and the intention behind it.

1. Why are we moving data from a const object?
2. If an optimization is possible/desirable why not add a `virtual future<> write(...) &&` overload?

Reply:
Also https://en.cppreference.com/w/cpp/language/const_cast notes: "Modifying a const object through a non-const access path and referring to a volatile object through a non-volatile [glvalue](https://en.cppreference.com/w/cpp/language/value_category#glvalue) results in undefined behavior."

Reply:
Good eye, this is definitely wrong and left over from my earlier experimentation: will fix.

Reply:
This is fixed in https://github.com/scylladb/seastar/commit/44e087a88b88002519672938e40535b6241524df